src/app/node-data/basic/provider/vmware-cloud-director/component.ts

```diff
@@ -0,0 +1,360 @@
+import {NodeProvider} from '@shared/model/NodeProviderConstants';
+import {NodeData} from '@shared/model/NodeSpecChange';
+import {BaseFormValidator} from '@shared/validators/base-form.validator';
+import {merge, Observable, Subject} from 'rxjs';
+import {filter, switchMap, take, takeUntil, tap} from 'rxjs/operators';
+
+enum Controls {
+  CPUs = 'cpus',
+  CPUCores = 'cpuCores',
+  MemoryMB = 'memoryMB',
+  DiskSizeGB = 'diskSizeGB',
+  DiskIOPs = 'diskIOPs',
+  IPAllocationMode = 'IPAllocationMode',
+  StorageProfile = 'storageProfile',
+  Catalog = 'catalog',
+  Template = 'template',
+}
+
+enum StorageProfileState {
+  Ready = 'Storage Profile',
+  Loading = 'Loading...',
+  Empty = 'No storage profiles available',
+}
+
+enum TemplateState {
+  Ready = 'Template',
+  Loading = 'Loading...',
+  Empty = 'No templates available',
+}
+
+enum CatalogState {
+  Ready = 'Catalog',
+  Loading = 'Loading...',
+  Empty = 'No catalogs available',
+}
+
+@Component({
+  selector: 'km-vmware-cloud-director-basic-node-data',
+  templateUrl: './template.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => VMwareCloudDirectorBasicNodeDataComponent),
+      multi: true,
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => VMwareCloudDirectorBasicNodeDataComponent),
+      multi: true,
+    },
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VMwareCloudDirectorBasicNodeDataComponent
+  extends BaseFormValidator
+  implements OnInit, AfterViewChecked, OnDestroy
+{
+  readonly Controls = Controls;
+  readonly ipAllocationModes = [VMwareCloudDirectorIPAllocationMode.POOL, VMwareCloudDirectorIPAllocationMode.DHCP];
+
+  @ViewChild('storageProfileCombobox')
+  private readonly _storageProfileCombobox: FilteredComboboxComponent;
+  @ViewChild('catalogCombobox')
+  private readonly _catalogCombobox: FilteredComboboxComponent;
+  @ViewChild('templateCombobox')
+  private readonly _templateCombobox: FilteredComboboxComponent;
+
+  storageProfiles: VMwareCloudDirectorStorageProfile[] = [];
+  catalogs: VMwareCloudDirectorCatalog[] = [];
+  templates: VMwareCloudDirectorTemplate[] = [];
+  selectedStorageProfile = '';
+  selectedCatalog = '';
+  selectedTemplate = '';
+  storageProfileLabel = StorageProfileState.Empty;
+  templateLabel = TemplateState.Empty;
+  catalogLabel = CatalogState.Empty;
+
+  private _catalogChanges = new Subject<boolean>();
+  private _datacenter: Datacenter;
+
+  constructor(
+    private readonly _builder: FormBuilder,
+    private readonly _nodeDataService: NodeDataService,
+    private readonly _clusterSpecService: ClusterSpecService,
+    private readonly _datacenterService: DatacenterService,
+    private readonly _cdr: ChangeDetectorRef
+  ) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this._initForm();
+
+    this._nodeDataService.nodeData = this._getNodeData();
+
+    this._datacenterService
```

Comment:
I think we should do that for all provider components in a separate PR otherwise this will be the odd one out. What I can do is move the form initialisation to `_initForm` method.

Reply:
sounds good to me!